

# -- system --
ctrl_period: 0.03               # control period [s]
traj_resample_dist: 0.1         # path resampling interval [m]
enable_yaw_recalculation: false # flag for recalculation of yaw angle after resampling
admisible_position_error: 5.0   # stop mpc calculation when error is larger than the following value
admisible_yaw_error_: 1.57      # stop mpc calculation when error is larger than the following value

# -- path smoothing --
enable_path_smoothing: false   # flag for path smoothing
path_filter_moving_ave_num: 25 # param of moving average filter for path smoothing 
curvature_smoothing_num: 15    # point-to-point index distance used in curvature calculation : curvature is calculated from three points p(i-num), p(i), p(i+num)

# -- mpc optimization --
qpoases_max_iter: 500                        # max iteration number for quadratic programming
qp_solver_type: "osqp"                       # optimization solver type. option is unconstraint_fast, unconstraint, and qpoases_hotstart and osqp
mpc_prediction_horizon: 50                   # prediction horizon step
mpc_prediction_dt: 0.1                       # prediction horizon period [s]
mpc_weight_lat_error: 0.1                    # lateral error weight in matrix Q
mpc_weight_heading_error: 0.0                # heading error weight in matrix Q
mpc_weight_heading_error_squared_vel: 0.3    # heading error * velocity weight in matrix Q
mpc_weight_steering_input: 1.0               # steering error weight in matrix R
mpc_weight_steering_input_squared_vel: 0.25  # steering error * velocity weight in matrix R
mpc_weight_lat_jerk: 0.0                     # lateral jerk weight in matrix R
mpc_weight_steer_rate: 0.0                   # steering rate weight in matrix R
mpc_weight_steer_acc: 0.000001               # steering angular acceleration weight in matrix R
mpc_weight_terminal_lat_error: 1.0           # terminal lateral error weight in matrix Q to improve mpc stability
mpc_weight_terminal_heading_error: 0.1       # terminal heading error weight in matrix Q to improve mpc stability
mpc_zero_ff_steer_deg: 0.5                   # threshold that feed-forward angle becomes zero

# -- vehicle model --
vehicle_model_type: "kinematics" # vehicle model type for mpc prediction. option is kinematics, kinematics_no_delay, and dynamics
input_delay: 0.24     # steering input delay time for delay compensation
vehicle_model_steer_tau: 0.3     # steering dynamics time constant (1d approzimation) [s]
steer_lim_deg: 40.0              # steering angle limit [deg]
steer_rate_lim_deg: 600.0        # steering angle rate limit [deg/s]

# -- lowpass filter for noise reduction --
steering_lpf_cutoff_hz: 3.0 # cutoff frequency of lowpass filter for steering command [Hz]
